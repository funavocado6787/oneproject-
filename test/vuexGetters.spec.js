import { RouterLinkStub, shallowMount, createLocalVue, mount } from '@vue/test-utils'
import VuexGetters from '../pages/vuexGetters'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Vuetify)
import DisplayCode from '../components/displayCode.vue'
Vue.component(DisplayCode.name, DisplayCode);

const localVue = createLocalVue()
localVue.use(Vuex)

describe('VuexGetters.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            vuexGettersSnippets: {}
        }
        store = new Vuex.Store({
            modules: {
                codeSnippet: {
                    namespaced: true,
                    actions: {
                        loadVuexGettersSnippets({commit}) {
                            commit('setVuexGettersSnippets', {
                                vuexGetters1: 'vuexGetters1',
                                vuexGetters2: 'vuexGetters2'
                             });
                        }
                    },
                    getters: {
                        getVuexGettersSnippets: state => state.vuexGettersSnippets
                    },
                    state,
                    mutations: {
                        setVuexGettersSnippets(state, data) {
                            state.vuexGettersSnippets = data;
                        }
                    }
                }
            }
        })
    })

    test('Smoke Test - check page renders by looking for some hard coded text', async () => {
        const wrapper = shallowMount(VuexGetters, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Vuex - Getters');
    })

    test('vuexGetters snippet load call was made on creation', async () => {
        const wrapper = shallowMount(VuexGetters, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Vuex - Getters');
        expect(Object.keys(state.vuexGettersSnippets).length).toBe(2);
    })

    test('vuexGetters made use of vuexGetters1', async () => {
        const wrapper = mount(VuexGetters, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('vuexGetters1');
    })

    test('vuexGetters made use of vuexGetters2', async () => {
        const wrapper = mount(VuexGetters, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('vuexGetters2');
    })

    test('match snapshot', async () => {
        const wrapper = mount(VuexGetters, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})













