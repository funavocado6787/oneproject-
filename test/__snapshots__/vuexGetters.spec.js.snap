// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VuexGetters.vue match snapshot 1`] = `
<div
  class="container"
>
  <h3>
    Vuex - Getters
  </h3>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          vuexGetters1
          <br />
        </pre>
      </div>
    </div>
  </div>
   
  <p>
    Mapping Getters on the component
  </p>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          vuexGetters2
          <br />
        </pre>
      </div>
    </div>
  </div>
</div>
`;
